### ISSUE TITLE
remote debugging when running server on different port

### ISSUE THREAD

**Issue by ConnorMWGraham** [facebook/react-native#12527]
### Description
I am running my server with "react-native start --port 9090". I can not use port 8081. Enabling Remote JS Debugging from my android emulator produces the following error:
![image](https://cloud.githubusercontent.com/assets/11632613/23228476/a425afa4-f90a-11e6-9a70-b8e5f0832cd6.png)
The error seems to be easy to understand. It is trying to load a resource from localhost:8081, but can't find it because my server is running on port 9090

### Reproduction
1. Start react-native server on different port "react-native start --port 9090"
2. Customize emulator to use port 9090 - ctrl+M > Dev Settngs > Debug server host & port for device >"10.0.2.2:9090"
3.  Enable remote debugging - ctrl+M > Debug JS Remotely

### Additional Information

* React Native version: 0.41.2
* Platform: Android
* Operating System: Windows
* Default Browser: Chrome

**Comment by brentvatne** [comment]
@ConnorMWGraham - did you rebuild the client using `react-native run-android` after changing the port?

**Comment by ConnorMWGraham** [comment]
No. I thought I had adjusted the client appropriately by doing ctrl+M > Dev Settngs > Debug server host & port for device >"10.0.2.2:9090". Where can I find the appropriate client configuration file?

**Comment by chmursonss** [comment]
Same issue here. I have to run packager on different port since 8081 is already taken by McAfee processes ( cannot change that easily :( )

**Comment by chmurson** [comment]
Same issue here. I have to run packager on different port since 8081 is already taken by McAfee processes ( cannot change that easily :( ). Everything works fine when I change host  & port device via "in-app menu" (CMD+M). However it returns same error as in issue description when I turn on the remote debugger.

**Comment by anders-g-hansen** [comment]
I believe that #12095 which has a PR that has not yet been merged will partially fix the issue.  I am able to work around this currently by setting a breakpoint in debuggerWorker.js and changing the port on the message.url before continuing (see screenshot).  This is a painful work around, but until that PR is merged, not sure there is a better option.

![debuggerworker js](https://cloud.githubusercontent.com/assets/11392648/23406526/ff011f2a-fd84-11e6-9c87-e3cb21020ff0.png)

However, I also needed to update the RealmReactModule.java file to change the hard coded port set for the Access-Control-Allow-Origin header to be the custom port I'm using (8088).  Otherwise I get CORS errors.

```
            Response response = newFixedLengthResponse(jsonResponse);
            response.addHeader("Access-Control-Allow-Origin", "http://localhost:8088");
            return response;
```

After doing these two things, I was able to get the remote debugging to work.
